File: system-tests-fixtures/android-container/lib/src/main/java/com/ernnavigation/ern/api/NavigationApi.java
/*
* Copyright 2017 WalmartLabs
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* http://www.apache.org/licenses/LICENSE-2.0
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/

package com.ernnavigation.ern.api;

import android.support.annotation.NonNull;

import com.walmartlabs.electrode.reactnative.bridge.ElectrodeBridgeEventListener;
import com.walmartlabs.electrode.reactnative.bridge.ElectrodeBridgeEvent;
import com.walmartlabs.electrode.reactnative.bridge.ElectrodeBridgeRequestHandler;
import com.walmartlabs.electrode.reactnative.bridge.ElectrodeBridgeResponseListener;
import com.walmartlabs.electrode.reactnative.bridge.None;
import java.util.*;
import java.util.UUID;


public final class NavigationApi {
    private static final Requests REQUESTS;

    static {
        REQUESTS = new NavigationRequests();
    }

    private NavigationApi() {
    }

    @NonNull
    public static Requests requests() {
        return REQUESTS;
    }



    public interface Requests {
        String REQUEST_NAVIGATE = "com.ernnavigation.ern.api.request.navigate";


        void registerNavigateRequestHandler(@NonNull final ElectrodeBridgeRequestHandler<NavigateData, Boolean> handler);

        void navigate(NavigateData navigateData, @NonNull final ElectrodeBridgeResponseListener<Boolean> responseListener);

    }
}

File: system-tests/fixtures/ios-container/ElectrodeContainer/APIs/NavigationAPI.swift
#if swift(>=4.0)
@objcMembers public class NavigationAPI: NSObject  {

    static let kRequestNavigate = "com.ernnavigation.ern.api.request.navigate";


    public lazy var requests: NavigationAPIRequests = {
        return NavigationRequests()
    }()
}


@objcMembers public class NavigationAPIRequests: NSObject {
    public func registerNavigateRequestHandler(handler: @escaping ElectrodeBridgeRequestCompletionHandler) -> UUID?{
        assertionFailure("should override")
        return UUID()
    }


    public func unregisterNavigateRequestHandler(uuid: UUID) -> ElectrodeBridgeRequestCompletionHandler? {
        assertionFailure("should override")
        return nil
    }


    public func navigate(navigateData: NavigateData, responseCompletionHandler: @escaping ElectrodeBridgeResponseCompletionHandler) {
        assertionFailure("should override")
    }

}
#else
public class NavigationAPI: NSObject  {

    static let kRequestNavigate = "com.ernnavigation.ern.api.request.navigate";


    public lazy var requests: NavigationAPIRequests = {
        return NavigationRequests()
    }()
}


public class NavigationAPIRequests: NSObject {
    public func registerNavigateRequestHandler(handler: @escaping ElectrodeBridgeRequestCompletionHandler) -> UUID?{
        assertionFailure("should override")
        return UUID()
    }


    public func unregisterNavigateRequestHandler(uuid: UUID) -> ElectrodeBridgeRequestCompletionHandler? {
        assertionFailure("should override")
        return nil
    }


    public func navigate(navigateData: NavigateData, responseCompletionHandler: @escaping ElectrodeBridgeResponseCompletionHandler) {
        assertionFailure("should override")
    }

}

#endif
